src/main/java/it/gov/pagopa/standinmanager/service/StationCalcService.java

```diff
@@ -92,53 +95,82 @@ public void runCalculations()
-                            try {
-                                eventHubService.publishEvent(ZonedDateTime.now(), station, Constants.type_removed);
-                            } catch (JsonProcessingException e) {
-                                log.error("could not publish {} for stations {}", Constants.type_removed, station);
-                                throw new RuntimeException(e);
-                            }
-                        }
-                        if (saveDB) {
-                            log.info("removing {} from standin database", station);
-                            dbStationsRepository.deleteById(station);
-                        }
-
-                        cosmosEventsRepository.newEvent(
-                                station,
-                                Constants.EVENT_REMOVE_FROM_STANDIN,
-                                String.format(
-                                        "removing station [%s] from standIn stations because %s calls were successful"
-                                                + " in the last %s minutes",
-                                        station, successfulCalls, rangeMinutes));
-
-                        String sendResult =
-                                awsSesClient.sendEmail(
-                                        String.format(
-                                                "[StandInManager][%s] Station [%s] removed from standin", env, station),
-                                        String.format(
-                                                "[StandInManager]Station [%s] has been removed from standin"
-                                                        + "\nbecause [%s] calls were successful in the last %s minutes",
-                                                station, successfulCalls, rangeMinutes));
-                        log.info("email sender: {}", sendResult);
+                    if (insertTime.plus(rangeMinutes, ChronoUnit.MINUTES).isBefore(Instant.now()) && successfulCalls > rangeLimit) {
+                        removeStationAutomatically(station, successfulCalls);
                     }
                 });
     }
+
+    public void removeStationFromStandIn(String station) {
+        List<CosmosStandInStation> stations = cosmosStationRepository.getStation(station);
+
+        if (stations.isEmpty()) {
+            throw new IllegalStateException("Station not found: " + station);
+        }
+
+        removeStationManually(station);
+    }
+
+    private void removeStationManually(String station) {
+        String eventInfo = String.format(
+                "removing station [%s] from standIn stations manually", station);
+
+        String emailBody = String.format(
+                "[StandInManager] Station [%s] has been removed from stand-in manually", station);
+
+        removeStation(station, eventInfo, emailBody);
+    }
+
+    private void removeStationAutomatically(String station, long successfulCalls) {
```

Comment:
Can you add a little comment to document the difference between automatic and manual removal?